src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -0,0 +1,204 @@
+            msg = f"Failed to create MSPileupObj, {exp}"
+            self.logger.exception(msg)
+            err = MSPileupError(pdict, msg)
+            return [err.DuplicateDocument()]
+
+        # insert document into underlying DB
+        try:
+            self.dbColl.insert_one(doc)
+        except errors.DuplicateKeyError:
+            msg = f"Failed to insert: {doc}, it already exist in DB"
+            self.logger.exception(msg)
+            err = MSPileupError(doc, msg)
+            return [err.DuplicateDocument()]
+        return []
+
+    def update(self, doc):
+        """
+        Update pileup data with provided input
+
+        :param doc: a dictionary of pieleup data to be updated
+        :return: list of MSPileupError or empty list
+        """
+        # check if given document contains pileup Name (unique key)
+        pname = doc.get('pileupName', '')
+        if not pname:
+            err = MSPileupError(doc, f'No document found for {pname}')
+            return [err.NoKeyFound()]
+
+        # look-up piileup document in underlying DB
+        spec = {'pileupName': pname}
+        results = self.get(spec)
+        if not results:
+            err = MSPileupError(spec, f'No document found for {spec} query')
+            return [err.NoDocumentFound()]
+
+        # at this point we should have single document corresponding to given pileup name
+        if len(results) != 1:
+            msg = f"Unique constrain violated for {pname}"
+            err = MSPileupError(spec, msg)
+            return [err.UniqueConstrainError()]
+
+        # we do not need to create MSPileupObj and validate it since our doc comes from DB directly
+        doc['lastUpdateTime'] = gmtime()
+        try:
+            self.dbColl.update_one(spec, {"$set": doc})
+        except Exception as exp:  # pylint: disable=broad-except
+            msg = f"Failed to insert: {doc}, error {exp}"
+            self.logger.exception(msg)
+            err = MSPileupError(doc, msg)
+            return [err.DatabaseError(exp)]
+        return []
+
+    def get(self, spec, projection=None):
+        """
+        fetches MSPileup data from persistent storage for a given spec (JSON query)
+
+        :param spec: input MongoDB query (JSON spec)
+        :param projection: MongoDB projection,
+        see https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html
+        :return: list of documents fetched from database
+        """
+        results = []
+
+        for doc in self.dbColl.find(spec, projection):
+            doc = stripKeys(doc, ['_id'])
+            results.append(doc)
+        return results
+
+    def delete(self, spec):
+        """
+        deletes MSPileup data in persistent storage for given spec (JSON query)
+
+        :param spec: MongoDB spec (JSON query)
+        :return: list of MSPileupError or empty list
+        """
+        try:
+            self.dbColl.delete_one(spec)
```

Comment:
How about adding a `self.logger.info("Pileup object %s successfully deleted from the database", spec.get("pileupName"))`?

Reply:
done